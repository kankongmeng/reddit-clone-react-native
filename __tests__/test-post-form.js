'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PostForm from '../app/components/post-form';

describe('PostForm', () => {
  let component;
  let Button;
  const defaultState = {
    value: {
      title: '',
      author: '',
      imageURL: 'https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/No_image_3x4.svg/1024px-No_image_3x4.svg.png',
      topic: '',          
    },
  };
    
  test('PostForm view renders correctly', () => {
    const tree = renderer.create(<PostForm />).toJSON()
    expect(tree).toMatchSnapshot();
  });

  beforeEach(() => {
    component = shallow(<PostForm />);
    Button = component.find('Button');
  });

  it('It has default state', () => {
    expect(component.state()).toEqual(defaultState);
  });

  it('It renders button with title', () => {
    const expectedTitle = 'Add Post';
    expect(Button.props().title).toEqual(expectedTitle);
  });

});

