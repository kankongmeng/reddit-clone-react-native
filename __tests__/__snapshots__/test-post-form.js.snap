// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm PostForm view renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "justifyContent": "center",
      "marginTop": 20,
      "padding": 15,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Title
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Title"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder="Title Cute Tiger"
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Author
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Author"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder="Alex Tan"
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Image url
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Image url"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/No_image_3x4.svg/1024px-No_image_3x4.svg.png"
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/No_image_3x4.svg/1024px-No_image_3x4.svg.png"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Topic
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Topic"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder="Tiger is cute and gorgeous..."
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Post
      </Text>
    </View>
  </View>
</View>
`;
